tests/test_hf_api.py

```diff
@@ -467,6 +467,83 @@ def test_get_full_repo_name(self):
         repo_name_with_no_org = self._api.get_full_repo_name("model", organization="org")
         self.assertEqual(repo_name_with_no_org, "org/model")
 
+    @use_tmp_repo()
+    def test_delete_files_r(self, repo_url: RepoUrl) -> None:
+        repo_id = repo_url.repo_id
+        fixtures = [
```

Comment:
In this case, I'd rather have a separate test per example. This way we ensure the tests are not dependent on eachother by using a different tmp repo each time.

Here a potential problem is that between each subTest we are deleting files to have a clean state. But since `delete_files` is actually the method we want to test, it might error out/have a weird behavior without us noticing.

Reply:
Actually, what I would do to test this:
1. have a separate TestCase class specific for `delete_files`
2. in `setUp`, create a repo + upload some files to it (always the same for each test).
3. for each test you want to try, write a separate test that calls `delete_files` with different args. Then you call `self._api.list_repo_files` and you check that only the files you wanted still exists. Testing `hf_hub_download` for that is a too heavy, it's preferable to list files without downloading.

(4. it is also important to check cases where deleting files will fail (say if files don't exist for example). These corner cases should be clearly identified, tested and documented.)

Reply:
Just implemented it as you'd like it :) see for yourself. 

Also added a new function to initialize a local temp folder. 
 I could also have done it that way ((pseudo code not-tested) : 
```python
def use_tmp_local_folder(self, test_fn: T) -> Callable[[T], T]:
    def _inner_use_local_folder(test_fn: T) -> T:
        @wraps(test_fn)
        def _inner(*args, **kwargs):
            if len(args) > 0 and isinstance(args[0], object):
                self:unittest.TestCase = args[0]
                if self.__class__ != unittest.TestCase:
                    raise TypeError(
                        f"Method {test_fn.__name__} can only be called from an instance of the 'unittest.TestCase' class"
                    )
            else:
                raise TypeError("This decorator can only be used on non-static methods of a 'unittest.TestCase' class.")

            self.tmp_dir = tempfile.mkdtemp()
            self.tmp_file = os.path.join(self.tmp_dir, "temp")
            self.tmp_file_content = "Content of the file"
            with open(self.tmp_file, "w+") as f:
                f.write(self.tmp_file_content)
            os.makedirs(os.path.join(self.tmp_dir, "nested"))
            self.nested_tmp_file = os.path.join(self.tmp_dir, "nested", "file.bin")
            with open(self.nested_tmp_file, "wb+") as f:
                f.truncate(1024 * 1024)

            self.addCleanup(rmtree_with_retry, self.tmp_dir)

            return test_fn(*args, **kwargs)

        return _inner

    return _inner_use_tmp_local_folder

```

Reply:
Hi @OlivierKessler01, I just pushed https://github.com/huggingface/huggingface_hub/pull/2280/commits/e953469ec57633eb530a1be50b369912142b563c to refacto a bit the tests. While the tests you wrote were valid, I've updated them for readability and simplicity. In particular:
- no need to create a local folder since what we want is to create a repo with files in it. This can be done directly with `create_commit`
- it's better to name tests with more explicit names than `test_one`, `test_two`, etc. This way it gives information for people reading those tests in the future.
- I've added more tests about nested folders and patterns.
- It's better to only test what the unit test is supposed to test. For instance, no need to check the files present in the repo after the first commit. If the upload commit fails (for a breaking change for instance), it will be covered by another test in the test suite. Usually you test 1 feature at a time in the unit test and assume all other methods are working correctly.  
- I feel that the `_check` method was hidding too many things to the reader. Sometimes it's better to explicitly test what you want to test . For example, testing only the fact that a file has been deleted rather than testing all files that should have remained.